### DISCUSSION TITLE
Missing trending activations like GEGLU & Mish

### DISCUSSION THREAD

**Discussion by HenryNdubuaku** [google/flax#3483] [Ideas] [answered]
GEGLU is a top activation right now but I had to implement it myself in a recent project. Is there a specific reason for omitting it? I can submit a PR for those?

**Comment by chiamp** [comment]
Flax [imports](https://github.com/google/flax/blob/main/flax/linen/activation.py#L24-L48) most of its activation functions from Jax. You could try submitting an issue/PR to Jax.

**Reply by HenryNdubuaku** [reply]
I noticed PReLu was crudely implemented in Flax’s activation, opposed to importing from Jax. This seemed premeditated, given that it has learnable parameters, just like GEGLU. Any idea on why?

**Reply by chiamp** [reply, marked as answer]
I think PReLU was implemented in Flax precisely because it has learnable parameters (and there is also [no Jax implementation](https://jax.readthedocs.io/en/latest/jax.nn.html#activation-functions) of it). In which case if GEGLU has  learnable parameters, maybe we can add it to Flax instead.

**Reply by HenryNdubuaku** [reply]
Great, I shall submit a PR.

**Reply by chiamp** [reply]
Great, thanks! Also I found the original [discussion](https://github.com/google/flax/issues/1419) for PReLU for context.

**Reply by chiamp** [reply]
And for Mish:
* https://github.com/google/flax/discussions/3391
* https://github.com/google/flax/issues/584
